Hawkbit dependencies clarification

FlorentRevest opened the pull request:
This makes a dependency rom Hawkbit to RabbitMQ more obvious.
More importantly, this also documents the exact steps of setting up a persistent database for Hawkbit. The official documentation was incomplete, lacking important steps such as an SQL request that has to be typed in manually.

FlorentRevest committed:
SOTA-System: Document Hawkbit's dependency to RabbitMQ

For some reason, Hawkbit depends on a daemon named "rabbitmq-server".
Hawkbit works without that daemon but produces many errors on its
command line output. It is better to list it as a dependency to make
sure we don't have those errors.

Signed-off-by: Florent Revest <revestflo@gmail.com>

FlorentRevest committed:
SOTA-System: Document the usage of Hawkbit with MariaDB

By default, Hawkbit keeps metadata and artifacts in memory. A paragraph
used to point to the official Hawkbit documentation on how to configure
Hawkbit to use a MariaDB backend.

However, the official documentation lacked some details and those
instructions should make the configuration much more straightforward and
less painful.

Signed-off-by: Florent Revest <revestflo@gmail.com>

frznlogic commented on docs/chapters/architecture/SOTA-system.rst:
```diff
@@ -570,13 +570,66 @@ see the status of your rollout campaign changed.
+        <scope>compile</scope>
+    </dependency>
+
+Finally, Hawkbit must be configured to connect to the database you created
```
Remove finally, imho

frznlogic reviewed (commented):
I looked at the document and kind of miss some kind of description of hawkbit. Not sure if I missed it somewhere or if it's in some other place?

FlorentRevest commented:
@frznlogic Hawkbit is first introduced in the "Solution strategy" section as the chosen SOTA Server. The concept of a "SOTA Server" is extensively described all along the document so I believe there is already enough.
In "Solution strategy", there is also a reference to an appendix that compares the various SOTA Clients and briefly introduces Hawkbit.

thenor commented on docs/chapters/architecture/SOTA-system.rst:
```diff
@@ -617,13 +617,66 @@ see the status of your rollout campaign changed.
+Persistent storage
+""""""""""""""""""
+
+Those instructions don't use a database to store artifacts and metadata. This
```
I know this is an old paragraph, but I think this might be a non-English way of referring to something. I don't understand what instructions are referenced here. "Those" implies to me that they were just mentioned (but there is no text above), but I would anyway have used "These" or "The following" or if referenced earlier "The above mentioned"

frznlogic commented:
@florentrevest I looked closer at that section now and I think it's unclearly stated and there's no real description of it. Anyways, I'll get back on it from a computer as I'm not sure if my phone is screwing me up somehow with the formatting etc (it looks really weird here).

thenor commented on docs/chapters/architecture/SOTA-system.rst:
```diff
@@ -617,13 +617,66 @@ see the status of your rollout campaign changed.
+
+Once MariaDB is running, you need to create a database for Hawkbit. For the
+rest of the instructions, we will use the default MariaDB user whose username
+is *root* and password is empty but you can create a new user and adapt the
```
Are we saying here that the recommended way of running MariaDB is with the user root?

FlorentRevest replied:
No! This is just the default user name and password of the DB itself.

thenor reviewed (commented):
I had just a comment on formatting and a question.

FlorentRevest commented:
@thenor: I addressed your comments
@frznlogic: I'll address your comment in another PR and merge this one right away because the instructions included here might be needed very soon.
